File: C++/2023_NumberOfPairsOfStringsWithConcat.cpp
class Solution {
public:
    int numOfPairs(vector<string>& nums, string target) {
        int output = 0, index = 0;
        while(index < nums.size()){
            int jIndex = 0;
            while(jIndex < nums.size()){
                if (index == jIndex){
                    jIndex++;
                    continue;
                }
                if (nums[index] + nums[jIndex] == target){
                    output += 1;
                }
                jIndex++;
            }
            index++;
        }
        return output;
    }
};

File: 2023_NumberOfPairsOfStringsWithConcat.py
class Solution:
    def numOfPairs(self, nums: List[str], target: str) -> int:
        output = 0
        index = 0
        while index < len(nums):
            jIndex = 0
            while jIndex < len(nums):
                if index == jIndex:
                    jIndex += 1
                    continue
                if nums[index] + nums[jIndex] == target:
                    output += 1
                jIndex += 1
            index += 1
        return output